tests/libcxx/CMakeLists.txt

```diff
@@ -12,6 +12,20 @@ add_subdirectory(host)
 
 if (BUILD_ENCLAVES)
     add_subdirectory(enc)
+elseif (LINUX_BIN_DIR)
+    # Find the build information from pre-compiled tests. If found, `LINUX_CXX_COMPILER_ID`
+    # and `LINUX_CXX_COMPILER_VERSION` will hold the corresponding information.
+    list(APPEND CMAKE_MODULE_PATH "${LINUX_BIN_DIR}/libcxx/enc")
+    include(build_info OPTIONAL)
+endif()
+
+if (ADD_WINDOWS_ENCLAVE_TESTS AND LINUX_CXX_COMPILER_ID AND LINUX_CXX_COMPILER_VERSION)
```

Comment:
Consider nesting the `LINUX_CXX_COMPILER_ID` and `LINUX_CXX_COMPILER_VERSION` checks under the `if (ADD_WINDOWS_ENCLAVE_TESTS)` case and explicitly erroring out if either are unset. This allows better trapping and diagnosis of an unexpected case than letting it fall through to using the CMAKE compiler properties and failing that way.